### PR TITLE
SDI-71: 📝 Convert to springdoc

### PR DESCRIPTION
**Pull request by petergphillips**
- SDI-71: 📝 Convert to springdoc
- SDI-71: 📝 Add in default 200 responses
- SDI-71: 📝 Add in produces to generate correct accept type
- SDI-71: 📝 Add in missing with annotation (previously removed wither)
- SDI-71: 📝 Fix docs test

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -109,7 +110,6 @@ tasks {
     }
     minHeapSize = "128m"
     maxHeapSize = "2048m"
-    jvmArgs = listOf("--add-opens", "java.base/java.lang=ALL-UNNAMED", "--add-opens", "java.base/java.util=ALL-UNNAMED")
```
Comment (petergphillips) [inline build.gradle.kts:112]: will be ignored in java 17 anyway

File: src/main/java/uk/gov/justice/hmpps/prison/PrisonApiServer.java [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+import org.springdoc.hateoas.SpringDocHateoasConfiguration;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.autoconfigure.domain.EntityScan;
 import org.springframework.scheduling.annotation.EnableScheduling;
-import springfox.boot.starter.autoconfigure.OpenApiAutoConfiguration;
 
 
-@SpringBootApplication(scanBasePackages = {"uk.gov.justice.hmpps"}, exclude = OpenApiAutoConfiguration.class)
+@SpringBootApplication(scanBasePackages = {"uk.gov.justice.hmpps"}, exclude = SpringDocHateoasConfiguration.class)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/PrisonApiServer.java:10]: was causing one of the tests to fail with missing dependency

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/AddressDto.java [hunk 3]
```diff
@@ -12,62 +11,62 @@
...
-@ApiModel(description = "An Address")
+@Schema(description = "An Address")
 @Data
 @Builder(toBuilder = true)
 @NoArgsConstructor
 @AllArgsConstructor
 @JsonInclude(JsonInclude.Include.NON_NULL)
 public class AddressDto {
 
-    @ApiModelProperty(value = "Address Id", example = "543524", position = 1)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/AddressDto.java:23]: `position` not supported

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/AddressDto.java [hunk 4]
```diff
@@ -12,62 +11,62 @@
...
+    @Schema(description = "Address Id", example = "543524")
     private Long addressId;
 
-    @ApiModelProperty(value = "Address Type", notes= "Reference domain is ADDR_TYPE", example = "BUS", position = 2)
+    @Schema(description = "Address Type. Note: Reference domain is ADDR_TYPE", example = "BUS")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/AddressDto.java:25]: had to move the `notes` into the `description` field

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/Agency.java [hunk 5]
```diff
@@ -12,41 +11,41 @@
...
-    @ApiModelProperty(required = true, value = "Agency description.", example = "Moorland (HMP & YOI)", position = 2)
+    @Schema(required = true, description = "Agency description.", example = "Moorland (HMP & YOI)")
     private String description;
 
-    @ApiModelProperty(value = "Long description of the agency", example = "Moorland (HMP & YOI)", position = 3)
+    @Schema(description = "Long description of the agency", example = "Moorland (HMP & YOI)")
     private String longDescription;
 
-    @ApiModelProperty(required = true, value = "Agency type.  Reference domain is AGY_LOC_TYPE", example = "INST", allowableValues = "CRC,POLSTN,INST,COMM,APPR,CRT,POLICE,IMDC,TRN,OUT,YOT,SCH,STC,HOST,AIRPORT,HSHOSP,HOSPITAL,PECS,PAR,PNP,PSY", position = 4)
+    @Schema(required = true, description = "Agency type.  Reference domain is AGY_LOC_TYPE", example = "INST", allowableValues = {"CRC","POLSTN","INST","COMM","APPR","CRT","POLICE","IMDC","TRN","OUT","YOT","SCH","STC","HOST","AIRPORT","HSHOSP","HOSPITAL","PECS","PAR","PNP","PSY"})
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/Agency.java:30]: `allowableValues` should be list instead under spring doc

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/ErrorResponse.java [hunk 6]
```diff
@@ -27,21 +26,21 @@
 @Data
 public class ErrorResponse {
 
-    @ApiModelProperty(required = true, value = "Response status code (will typically mirror HTTP status code).", example = "404", allowableValues = "400-", position = 1)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/ErrorResponse.java:30]: these `allowableValues` made no sense

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/BookingAdjustment.java [hunk 7]
```diff
@@ -9,13 +9,13 @@
...
 
-@Schema(description = "Sentence Adjustment values")
+@Schema(description = "Booking Adjustment values")
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
 @Data
 public class BookingAdjustment {
-    @Schema(description = "Adjustment type", allowableValues = {"SPECIAL_REMISSION","ADDITIONAL_DAYS_AWARDED","RESTORED_ADDITIONAL_DAYS_AWARDED","UNLAWFULLY_AT_LARGE","LAWFULLY_AT_LARGE"})
+    @Schema(description = "Adjustment type")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/BookingAdjustment.java:18]: springdoc automatically generates `allowableValues` if enum type anyway.  They were appearing as duplicates when `allowableValues` specified

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**